src/main/java/juja/microservices/users/service/UserService.java

```diff
@@ -65,15 +86,63 @@ public UserService(UserRepository userRepository) {
                 .collect(Collectors.toList());
         logger.debug("Received response from repository: {}", users.toString());
 
-        List<UserDTO> result = users.stream()
-                .map(this::convertUserToUserDto)
-                .collect(Collectors.toList());
+        List<UserDTO> result = getConvertedResult(users);
         logger.debug("All users converted: {}", result.toString());
 
         return result;
     }
 
-    private UserDTO convertUserToUserDto(User user) {
-        return new UserDTO(user.getUuid(), user.getSlack(), user.getSkype(), user.getFullName());
+    @Scheduled(cron = "${cron.expression}")
+    public void scheduleUpdateUsers() {
+        updateUsersFromCRM();
+    }
+
+    public List<UserDTO> updateUsersFromCRM() {
+        Long lastUpdate = repository.findMaxLastUpdate();
+        if (lastUpdate == null) lastUpdate = 0L;
+
+        logger.info("Starting update users database. Last update was at {}",
+                LocalDateTime.ofEpochSecond(lastUpdate, 0, OffsetDateTime.now().getOffset()));
+
+        return getConvertedResult(updateUsersDatabase(getUpdatedUsersFromCRM(lastUpdate)));
```

Comment:
short form of writing could be the resource of errors that is difficult to find and read code. Too long chain of invocation several methods. Please write separately.

Reply:
Done! But I think my new version looks like a "Pascal style" :)